src/index.ts

```diff
@@ -415,6 +415,36 @@ function shouldIgnore (filename: string, ignore: RegExp[]) {
   return ignore.some(x => x.test(relname))
 }
 
+/**
+ * unsafe re-order and make .ts > .js
+ */
+function registerHandler (
+  opts: Options,
+  extensions: string[],
+  ignore: RegExp[],
+  register: Register,
+  originalJsHandler: (m: NodeModule, filename: string) => any
+) {
+  if (!opts.preferTsExts) {
+    return extensions.forEach(ext => registerExtension(ext, ignore, register, originalJsHandler))
+  }
+
+  // @todo a better way with options
+  ['.ts', '.tsx']
+    .concat(Object.keys(require.extensions), extensions) // tslint:disable-line
```

Comment:
This doesn't seem necessary, `extensions` already contains the extensions we care about and in the correct order. What if we just do a loop directly on that and make this flag `--reorder-exts` (since `--reorder-ts-exts` is not true in this PR, it may contain `.js`.

Reply:
Sorry, or `preferExts` if that sounds reasonable. Forgot the flag name 😄

Reply:
I think it should either be `--reorder-exts` or `--prefer-ts-exts`, since `prefer extensions` doesn't really mean anything by itself (i.e you're prefering to use extensions..?)

Thats also why I suggested `--prefer-ts-exts`, as I felt it gave you enough useful information about what it does - it reorders the order of extensions so that `.ts` extensions are "prefered" in situations where there is both a `.ts` & `.js` file with the same name.

But so if you'd prefer "reorder", then I'll make it `--reorder-exts` 😄

Reply:
Ah, I think I see - because of the concat order it doesn't actually re-order `.js` and `.jsx`. `--prefer-ts-exts` makes sense then. I'd maybe skip over the select, delete, assign if the property doesn't exist to start with to avoid setting `.tsx = undefined`.

Reply:
> I'd maybe skip over the select, delete, assign if the property doesn't exist to start with to avoid setting .tsx = undefined.

So, what change should I make...?

I tried changing the part at the bottom, the:

```
      const old = require.extensions[ext] // tslint:disable-line
      delete require.extensions[ext] // tslint:disable-line
      require.extensions[ext] = old // tslint:disable-line
```

but it for some reason breaks without it. I can't figure out why :/ 
Like I compared the before & after values, and they're apparently equal and everything, but for some reason if we don't select, delete & assign it doesn't work.